hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -574,14 +574,18 @@ public void cleanup(boolean invalidateClient) {
    * @throws IOException if stream is closed
    */
   private void checkOpen() throws IOException {
-    if (xceiverClient == null) {
+    if (isClosed()) {
       throw new IOException("BlockOutputStream has been closed.");
     } else if (getIoException() != null) {
       adjustBuffersOnException();
       throw getIoException();
     }
   }
 
+  public boolean isClosed() {
```

Comment:
This need not be public. If you need it for testing, please annotate.

Reply:
This needs to be pulic because this function gets invoked from BlockoutputStreamEntryPool#isClosed() and they are in different packages.